### ISSUE TITLE
Prevent DYNAMITE from repeatedly trying to solve unsolvable models (Meng's issue)

### ISSUE THREAD

**Issue by maindlt** [dynamics-of-stellar-systems/dynamite#256]
Implement a fix along the lines of
<img width="288" alt="image" src="https://user-images.githubusercontent.com/8792350/193541942-b7e4d961-e573-4514-b5b0-53ff0039534a.png">

**Comment by momomaycry** [comment]
I also find a bug after removing empty orblibs. I add to lines to in the 

```
for row in rows_orblib:
    self.logger.info(f'new orblib {orblib_dir}’)
for row in rows_ml:
   self.logger.info(f'orblib for ml {orblib_dir}')
```

Then I got the following output: 

> [INFO] 10:48:05 - dynamite.model.AllModels - Model directory orblib_022_005/ removed.
> [INFO] 10:48:06 - dynamite.model.AllModels - Model directory orblib_012_002/ removed.
> [INFO] 10:48:06 - dynamite.config_reader.Configuration - Instantiated AllModels object
> [INFO] 10:48:06 - dynamite.model_iterator.ModelIterator - LegacyGridSearch: iteration 35
> [INFO] 10:48:13 - dynamite.parameter_space.LegacyGridSearch - LegacyGridSearch added 36 new model(s) out of 53
> [INFO] 10:48:14 - dynamite.model_iterator.ModelInnerIterator - orblib for ml orblib_019_000
> [INFO] 10:48:14 - dynamite.model_iterator.ModelInnerIterator - orblib for ml orblib_019_000
> [INFO] 10:48:14 - dynamite.model_iterator.ModelInnerIterator - orblib for ml orblib_022_005
> ......
> [INFO] 10:48:14 - dynamite.model_iterator.ModelInnerIterator - orblib for ml orblib_022_005
> [INFO] 10:48:14 - dynamite.model_iterator.ModelInnerIterator - orblib for ml orblib_022_005

orblib_022_005/ was removed, but it didn’t appear as a new orblib, but only as a new ml, therefore several new ml will solve the orblib for orblib_022_005/ together, and caused conflict errors, as I submitted in **Issue 257**
**I suppose it is because the all_models used in self.is_new_orblib() were not updated after removing those empty models.**

> [INFO] 10:52:55 - dynamite.orblib.LegacyOrbitLibrary - Calculating initial conditions
> [ERROR] 10:52:58 - dynamite.orblib.LegacyOrbitLibrary - cmd_orb_start exit code 2. ERROR. Message: At line 194 of file orbitstart.f90 (unit = 31)
> Fortran runtime error: Cannot open file 'datfil/begin.dat': File exists
> Logfile: /home/myang/8138-6102/models/orblib_022_005/datfil/orbstart.log.

**Comment by maindlt** [comment]
> This is because of I included try except as this,
> 
> ```
> orblib = mod.get_orblib()
> orb_done = True
> if get_weights:
>                 try:
>                     weight_solver = mod.get_weights(orblib)
>                     wts_done = True
>                 except:
>                     mod.chi2, mod.kinchi2, mod.kinmapchi2 = np.inf, np.inf, np.nan
>                     wts_done = True
> ```

Hm, strange, a similar try-except (which takes care of failed orblib and failed weights) was introduced in PR #201...

**Comment by maindlt** [comment]
Hello @momomaycry,

I found a bug that may result in the all_models table being inconsistent with the orblibs on disk in case `reattempt_failures: False` in the config file (this somehow slipped through our testing...). Exactly this happens in the example you emailed to us and it may explain the strange behavior because the `orblib_done`, `weights_done`, and `all_done` indicators don't reflect what data is on the disk and what's missing...

Please try the quick fix in PR #262. In my tests your example runs (until I get an out of memory error later, but that's due to the small cluster queue I use for testing)... Let me know please ;-)

Perhaps we should also implement a utility method to ensure that `orblib_done` and `weights_done` are only `True` if the data is on the disk...

All the best,
Thomas
